static/app/components/events/interfaces/uptime/uptimeDataSection.tsx

```diff
@@ -77,36 +101,84 @@ export function DowntimeDuration({group}: {group: Group}) {
   const alertRuleId = event.tags.find(tag => tag.key === 'uptime_rule')?.value;
 
+  const elementRef = useRef<HTMLDivElement>(null);
+  const {width: containerWidth} = useDimensions<HTMLDivElement>({elementRef});
+  const timelineWidth = useDebouncedValue(containerWidth, 500);
+  const timeWindow = location.query?.timeWindow as TimeWindow;
+  const {since, until} = usePageFilterDates();
+
+  const timeWindowConfig = useMemo(() => {
+    if (defined(timeWindow)) {
+      const {start, end} = getTimeRangeFromEvent(event, now, timeWindow);
+      return getConfigFromTimeRange(start, end, timelineWidth);
+    }
+    return getConfigFromTimeRange(since, until, timelineWidth);
+  }, [timeWindow, timelineWidth, since, until, event, now]);
+
+  const {data: uptimeStats, isPending} = useUptimeMonitorStats({
+    ruleIds: alertRuleId ? [alertRuleId] : [],
+    timeWindowConfig,
+  });
+  const bucketedData = alertRuleId ? uptimeStats?.[alertRuleId] ?? [] : [];
+
+  const actions = (
+    <ButtonBar gap={1}>
+      {defined(alertRuleId) && (
+        <LinkButton
+          icon={<IconSettings />}
+          size="xs"
+          to={normalizeUrl(
+            `/organizations/${organization.slug}/alerts/rules/uptime/${project.slug}/${alertRuleId}/details/`
+          )}
+        >
+          {t('Uptime Alert Rule')}
+        </LinkButton>
+      )}
+      <ResolutionSelector />
+    </ButtonBar>
+  );
+
   return (
     <InterimSection
       title={t('Downtime Information')}
       type={SectionKey.DOWNTIME}
       help={t('Information about the detected downtime')}
       preventCollapse
-      actions={
-        alertRuleId !== undefined && (
-          <LinkButton
-            icon={<IconSettings />}
-            size="xs"
-            to={normalizeUrl(
-              `/organizations/${organization.slug}/alerts/rules/uptime/${project.slug}/${alertRuleId}/details/`
-            )}
-          >
-            {t('Uptime Alert Rule')}
-          </LinkButton>
-        )
-      }
+      actions={actions}
     >
-      {isResolved
-        ? tct('Domain was down for [duration]', {
-            duration: <DowntimeDuration group={group} />,
-          })
-        : tct('Domain has been down for [duration]', {
-            duration: <DowntimeDuration group={group} />,
-          })}
+      <Text>
+        {isResolved
+          ? tct('Domain was down for [duration]', {
+              duration: <DowntimeDuration group={group} />,
+            })
+          : tct('Domain has been down for [duration]', {
+              duration: <DowntimeDuration group={group} />,
+            })}
+      </Text>
+      <TimelineContainer>
+        <TimelineWidthTracker ref={elementRef} />
+        <StyledGridLineTimeLabels timeWindowConfig={timeWindowConfig} />
+        <GridLineOverlay timeWindowConfig={timeWindowConfig} showCursor={!isPending} />
+        {bucketedData && !isPending ? (
+          <FadeInContainer>
+            <CheckInTimeline
```

Comment:
seeing some weird behavior with the timeline when i make screen smaller where it overflows into the sidebar (though this is prob a bigger problem for a diff PR)

Reply:
Yeah, that comes from debouncing the resize. It should only last a few second or two before requerying for an appropriate amount of data in the chart.